src/components/buttons/ProgressButton.js

```diff
@@ -1,21 +1,36 @@
-import React, { useState } from 'react'
-import { View, StyleSheet } from 'react-native'
+import React, { useState, useEffect, useRef } from 'react'
+import { View, StyleSheet, Animated } from 'react-native'
 import { PrimaryButton, GrayButton } from '~/components'
 import { theme } from '~/constants'
 
 const ProgressBar = ({ progress }) => {
+  const progressAnimated = useRef(new Animated.Value(0))
+
+  useEffect(() => {
+    Animated.timing(progressAnimated.current, {
+      toValue: progress,
+      duration: 300
+    }).start()
+  }, [progress])
```

Comment:
You should stop the animation before starting a new one. The `useEffect` API allows to do that easily:
```javascript
useEffect(() => {
  const animation = Animated.timing(progressAnimated.current, {
    toValue: progress,
    duration: 300
  })
  
  animation.start()

  return () => animation.stop()
}, [progress])
```

Reply:
For what I understand there's no need to stop animations in this case. Animations get interrupted when others animations take place. Otherwise I think that would be visible in the progressBar.